src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -230,63 +218,57 @@ public async Task GetInfoPackage()
+            try
             {
-                try
-                {
-                    if (Debugger.IsAttached)
-                        throw new InvalidOperationException("Cannot get stack traces when debugger is attached");
+                var ms = new MemoryStream();
+                var isInfoEnabled = LoggingSource.Instance.IsInfoEnabled;
+                var isOperationsEnabled = LoggingSource.Instance.IsOperationsEnabled;
 
-                    ThreadsHandler.OutputResultToStream(sw);
+                if (isInfoEnabled == false)
+                    LoggingSource.Instance.IsInfoEnabled = true;
+                if (isOperationsEnabled == false)
+                    LoggingSource.Instance.IsOperationsEnabled = true;
 
-                    var result = JObject.Parse(sw.GetStringBuilder().ToString());
+                LoggingSource.Instance.AttachPipeSink(ms);
+                var sp = Stopwatch.StartNew();
 
-                    var wait = 100 - sp.ElapsedMilliseconds;
-                    if (wait > 0)
-                    {
-                        // I expect this to be _rare_, but we need to wait to get a correct measure of the cpu
-                        Thread.Sleep((int)wait);
-                    }
+                while (sp.ElapsedMilliseconds < 15000)
+                {
+                    await Task.Delay(3000);
+                }
 
-                    var threadStats = threadsUsage.Calculate();
-                    result["Threads"] = JArray.FromObject(threadStats.List);
+                LoggingSource.Instance.DetachPipeSink();
 
-                    using (var writer = new StreamWriter(stackTraceStream))
-                    {
-                        result.WriteTo(new JsonTextWriter(writer) { Indentation = 4 });
-                        writer.Flush();
-                    }
-                }
-                catch (Exception e)
-                {
-                    var jsonSerializer = DocumentConventions.Default.CreateSerializer();
-                    jsonSerializer.Formatting = Formatting.Indented;
+                if (isInfoEnabled == false)
+                    LoggingSource.Instance.IsInfoEnabled = false;
+                if (isOperationsEnabled == false)
+                    LoggingSource.Instance.IsOperationsEnabled = false;
 
-                    using (var errorSw = new StreamWriter(stackTraceStream))
-                    {
-                        jsonSerializer.Serialize(errorSw, new
-                        {
-                            Error = e.Message
-                        });
-                    }
+                ms.Position = 0;
+
+                var entry = archive.CreateEntry(prefix, CompressionLevel.Optimal);
+                entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
 
+                using (var entryStream = entry.Open())
+                {
+                    await ms.CopyToAsync(entryStream);
```

Comment:
do we dispose the `ms`? also, can't we pass the `entryStream` from the first place?

Reply:
done